### PR TITLE
gh-100062: Remove error code tables from _ssl and err_names_to_codes

### PR DESCRIPTION
**Pull request by davidben**
Prior to https://github.com/python/cpython/pull/25300, the make_ssl_data.py script used various tables, exposed in _ssl, to update the error list.

After that PR, this is no longer used. Moreover, the err_names_to_codes map isn't used at all. Clean those up. This gets them out of the way if, in the future, OpenSSL provides an API to do what the code here is doing directly. (https://github.com/openssl/openssl/issues/19848)

* Issue: gh-100062

### CODE REVIEW

File: Modules/_ssl.c
Comment (davidben) [inline Modules/_ssl.c:6180]: I'm not very familiar with the CPython API, so I wasn't sure if this should be removed now that the objects are no longer exported. I noticed `Sock_Type` doesn't seem to be exported, but is traversed, while `str_library` is also unexported but not traversed.

**Comment by davidben** [comment]
@tiran This PR look reasonable? Anything missing on my end?

**Comment by arhadthedev** [comment]
Adding the `skip news` label because `_ssl.h` is located outside `Include` and thus cannot be used by third parties.

**Comment by arhadthedev** [comment]
@jackjansen, @dstufft, @alex (as active ssl [experts](https://devguide.python.org/core-developers/experts/))

**Comment by alex** [comment]
This looks good to me, but I'm going to wait for a bit before merging in case one of the other maintainers know of a reason these were exposed.

**Comment by davidben** [comment]
My (possibly wrong) guess is that were exposed for the older version of the make_ssl_data.py script.

File: Modules/_ssl.c
Comment (zooba) [reply Modules/_ssl.c:6180]: This is for cycle detection. `Sock_Type` is a type object, which may contain references that could cause cycles, while I assume `str_library` is a string, which can never participate in a cycle. So we visit the former but not the latter.

**Review by zooba — APPROVED** [review]
LGTM

**Comment by zooba** [comment]
Let's merge this before the next alpha so there's more time for someone to shout if we shouldn't have dropped it.

**Pull request merged**
